consensus/src/detail/tests/processing_tests.rs

```diff
@@ -0,0 +1,242 @@
+        assert!(consensus.get_best_block_id().unwrap().is_some());
+        assert!(
+            consensus.get_best_block_id().ok().flatten().unwrap()
+                == consensus.chain_config.genesis_block().get_id()
+        );
+        assert!(consensus
+            .blockchain_storage
+            .get_block(consensus.chain_config.genesis_block().get_id())
+            .unwrap()
+            .is_some());
+        assert!(
+            consensus
+                .blockchain_storage
+                .get_block(consensus.chain_config.genesis_block().get_id())
+                .unwrap()
+                .unwrap()
+                .get_id()
+                == consensus.chain_config.genesis_block().get_id()
+        );
+    });
+}
+
+#[test]
+fn test_spend_inputs_simple() {
+    common::concurrency::model(|| {
+        let mut consensus = setup_consensus();
+
+        // Create a new block
+        let block = produce_test_block(
+            &consensus.chain_config,
+            consensus.chain_config.genesis_block(),
+            false,
+        );
+
+        // Check that all tx not in the main chain
+        for tx in block.transactions() {
+            assert!(
+                consensus
+                    .blockchain_storage
+                    .get_mainchain_tx_index(&OutPointSourceId::from(tx.get_id()))
+                    .expect(ERR_STORAGE_FAIL)
+                    == None
+            );
+        }
+
+        // Process the second block
+        let new_id = Some(block.get_id());
+        assert!(consensus.process_block(block.clone(), BlockSource::Local).is_ok());
+        assert_eq!(
+            consensus
+                .blockchain_storage
+                .get_best_block_id()
+                .expect(ERR_BEST_BLOCK_NOT_FOUND),
+            new_id
+        );
+
+        // Check that tx inputs in the main chain and not spend
+        for tx in block.transactions() {
+            let tx_index = consensus
+                .blockchain_storage
+                .get_mainchain_tx_index(&OutPointSourceId::from(tx.get_id()))
+                .expect("Not found mainchain tx index")
+                .expect(ERR_STORAGE_FAIL);
+
+            for input in tx.get_inputs() {
+                if tx_index
+                    .get_spent_state(input.get_outpoint().get_output_index())
+                    .expect("Unable to get spent state")
+                    != OutputSpentState::Unspent
+                {
+                    panic!("Tx input can't be spent");
+                }
+            }
+        }
+    });
+}
+
+#[test]
+fn test_straight_chain() {
+    common::concurrency::model(|| {
+        const COUNT_BLOCKS: usize = 255;
+        // In this test, processing a few correct blocks in a single chain
+        let config = create_mainnet();
+        let storage = Store::new_empty().unwrap();
+        let mut consensus = Consensus::new_no_genesis(config, storage).unwrap();
```

Comment:
Oh, it's because we can have an empty consensus? What is the use case for that?

Reply:
For testing + we will have different genesis blocks for different chain configs (testnet genesis shouldn't be the same as mainnet genesis).